test/fail_compilation/bug19225.d

```diff
@@ -0,0 +1,16 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/bug19225.d(15): Error: basic type expected, not `else`
+fail_compilation/bug19225.d(15):        There's no `static else`, use `else` instead.
+fail_compilation/bug19225.d(15): Error: no identifier for declarator `_error_`
```

Comment:
Are error messages containing `_error_` supposed to be printed?

Reply:
I don't know if they're supposed to. But those lines also show up without my patch.
see:
```
$ rdmd --compiler=ldc2 --eval="static if(false){}static else if {}"
(18): Error: basic type expected, not else
(18): Error: no identifier for declarator _error_
(18): Error: found else instead of statement
(19): Error: unrecognized declaration
```

Reply:
(To avoid confusion, `ldc` targets to LDC v1.9.0 on my machine.)

Reply:
I general it is good to only a one error and hence error message for one error in the code so as to avoid cascaded errors like this.

this is usually fixed by adding a 
```
if (foo.tok == TOK.error) return; /*or otherwise give up further parsing/semantic analysis*/
else checnkTheCondition();
```

Reply:
(but also with DMD, the `_error_` thingy is still there)

Reply:
@thewilsonator How would I implement this in this case?
Everything that happens next to the error message is returning `AST.Type.terror`.

Reply:
add a guard like [this]( https://github.com/dlang/dmd/blob/4a2805d4986ee85899abeb7903f523526603291c/src/dmd/parse.d#L3116) to [here](https://github.com/dlang/dmd/blob/4a2805d4986ee85899abeb7903f523526603291c/src/dmd/parse.d#L4562)